src/lib/transit/planning/route-finder.ts

```
@@ -232,70 +375,322 @@ export async function findAllTransferRoutes(
+
+      // Skip if stations are the same (unlikely but possible)
+      if (origin.station.id === destination.station.id) continue;
+
+      // Check if the stations share any lines for direct routes
+      const commonLines = origin.lines.filter((l1) =>
+        destination.lines.some((l2) => l2.id === l1.id)
+      );
+
+      // Try direct route first if stations share a line
+      if (commonLines.length > 0) {
+        for (const line of commonLines) {
```

ESLint

[error] 'line' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)